### PR TITLE
default the ignorenotfound for delete when selecting objects

### PR DESCRIPTION
**Pull request by deads2k**
When selecting resources and then deleting them (as opposed to specifically indicating a resource to delete) we should default to not failing if something disappeared partway through.  This is quite common when label selecting deletes and GC runs before you get to a dependent object.

I also took the liberty of doing complete work in complete instead of validate.

```release-note
`kubectl delete` with selection criteria defaults to ignoring not found errors
```

@kubernetes/sig-cli-maintainers

### CODE REVIEW

File: pkg/kubectl/cmd/delete.go [hunk 1]
```diff
@@ -185,40 +211,10 @@ func (o *DeleteOptions) Complete(f cmdutil.Factory, out, errOut io.Writer, args
 }
 
 func (o *DeleteOptions) Validate(cmd *cobra.Command) error {
-	if o.DeleteAll && len(o.LabelSelector) > 0 {
```
Comment (liggitt) [inline pkg/kubectl/cmd/delete.go]: these still belong in Validate()

**Comment by liggitt** [comment]
nit on validation move, and tests are red, lgtm otherwise

**Comment by deads2k** [comment]
nit fixed

**Review by juanvallejo — APPROVED** [review]

**Review by soltysh — APPROVED** [review]

**Pull request merged**